src/qibolab/instruments/qblox/controller.py

```diff
@@ -94,6 +97,22 @@ def disconnect(self):
             self.cluster.disconnect()
             self.is_connected = False
 
+    def _get_module_channel_map(self, module_name: str, qubits: dict):
+        """Retrieve all the channels connected to a specific Qblox module.
+
+        This method updates the `channel_port_map` attribute of the specified Qblox module
+        based on the information contained in the provided qubits dictionary (dict of `qubit` objects).
+
+        Return the list of channels connected to module_name"""
+        module = self.modules[module_name]
+        channels = []
+        for qubit in qubits.values():
+            for channel in qubit.channels:
+                if channel.port and channel.port.module.name == module_name:
+                    module.channel_port_map[channel.name] = channel.port.name
```

Comment:
Thanks for the explanations @PiergiorgioButtarini. 

I have not checked in much detail so I may be wrong, but just by looking at this line I think you don't need cycles to extract the information stored in `channel_port_map`. You are mapping the `channel.name` to the `port.name`, but since our current hierarchy is also `Qubit -> Channel -> Port`, if you have a `channel` you can directly do `channel.port.name` to get the port.

I think the cycles are needed for the inverse of this: if I give you a port, which channel is it connected to it.

In practice, for `get_if`, if instead of the `channel_port_map` which is string to string, you had a global channel map from the channel name to the actual `Channel` object, you could just do:
```py
_lo = self.channel_map[pulse.channel].lo_frequency
```

Regarding the `qubits`, indeed in `get_if` you don't have direct access to them, however `get_if` is mostly called by `process_pulse_sequence`, which has the `qubits`, so you could pass them to `get_if` if that simplifies your life. Before pushing towards that, I would like to discuss with others involved in drivers (potentially next week) if they would prefer a `Qubit`-centric or `Channel`-centric direction. Now it's a bit confusing because some drivers to one, others do the other. We should certainly stick to one, but I am not sure which.

Reply:
Thanks for the suggestion @stavros11, I've implemented it and as soon as the machines will be again available I will test it. 
By the way my concern was about if it's better to keep an additional attribute (`channel_port_map` or the more convenient `channel_map`) or passing in the `get_if` method the `qubits` dictionary. 
I think the former solution is better especially because `get_if` is also called in `_process_acquisition_results` that has no access to the `qubits`, so I will keep the global `channel_map`.

In fact even considering to pass `qubits` in the `get_if` method I can't see a way of extracting `lo_frequency` without two for loops.
Maybe it's easier to discuss these in person tomorrow if you have time

Reply:
> In fact even considering to pass `qubits` in the `get_if` method I can't see a way of extracting `lo_frequency` without two for loops.

That could be something like
```py
channel_type = pulse.type.name.lower() # or something like that
_lo = getattr(qubits[pulse.qubit], channel_type).lo_frequency
```
which I believe is used in some other drivers. Here `channel_type` can be one of `{"readout", "drive", "flux"}`.
However, I would not advise going into this direction until we are sure we want to use qubits in the drivers. Currently qblox is more channel oriented and this may be better overall

> I think the former solution is better especially because `get_if` is also called in `_process_acquisition_results` that has no access to the `qubits`, so I will keep the global `channel_map`.

Indeed, I just didn't mention it in the previous comment because I think I am dropping the `_process_acquisition_results` completely in #648, so that could solve this issue (although I guess we would still need to pass `qubits` somewhere). But as I said above it is probably better to stick to the `channel_map` for the purpose of this PR.

Reply:
> That could be something like
>  ``` python
>   channel_type = pulse.type.name.lower() # or something like that
>   _lo = getattr(qubits[pulse.qubit], channel_type).lo_frequency
  ```

Oh I see, thank you very much!

> But as I said above it is probably better to stick to the `channel_map` for the purpose of this PR.

Ok perfect.